Space dragon will create a plasma fire when they use their fire breath.

Pickle-Coding opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Space dragons will release plasma and oxygen at 2,000 Kelvin on every turf their fire goes through. The space dragon will store some plasma, up to a maximum of 800 moles, release up to 400 moles of plasma in a single fire breath, and generates 20 moles of plasma per second. The plasma storage thing is to prevent people from having to spam fire breaths in order to achieve optimal plasma fires. 0.4 moles of oxygen gets released per mole of plasma, giving a stoichiometric burn and allowing the plasma to burn in the room instead of spreading too far if too much oxygen got burned. It's not enough plasma to create a station wide plasma flood, and won't be able to spread too far.

Gives space dragon and space carps full heat and pressure resistance, so their plasma fires won't kill themselves or push themselves around.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
### Plasma.
It's a dragon. They should be scary and destructive, filling the room they are in in flames. The chaotic nature of local plasma fires will make the antagonist feel more impactful and the fight more intense.
### Event weight to 5 from 7.
A maintainer said something about dynamic. The dragon got buffed significantly, and carps can live in fire, so it might as well be a rater but more memorable event.
### Dragon and carp fire and pressure immunity.
Well the dragon shoots flaming plasma, which will heat the room and raises pressure. It would suck if the dragon's firebreath killed itself and their carps.
### Dragon firebreath team kill prevention.
Encourages the carps to work with the dragon instead of flying away to grief the station randomly.
### Dragon firebreath blocking generalisation.
Hardcoding random objects to block firebreath is lame. It makes sense to generalise it to whether atmos can pass or not (with the exception of space, because space it weird and blocks atmos).
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: Space dragons will now release small plasma fires when they breathe fire.
balance: Space dragons and space carps are now immune to heat, and resist space wind.
balance: Space dragon fire stream will no longer ignore armour, and will deal 10 brute and 20 burn damage.
balance: Reduce space dragon event weight to 5, from 7.
balance: Space dragons will no longer cause friendly fire on carps that are hit by their fire breath.
balance: Space dragon fire breath will now get blocked by anything that blocks atmos, instead of just closed turfs and shut doors and windows.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Pickle-Coding committed:
Greetings.

Adds get_stream_turfs proc for the space dragon. It returns the turfs in a line with a length of 20, or shorter if a closed turf, window, or door with a density blocks it.

Fire stream will iterate through stream_turfs, which is a list of turfs from get_stream_turfs.

Adds defines for space dragon plasma related values.

Undefines those defines at the end of the file.

dragon_fire_line() will create plasma and oxygen if the turf is an open turf.

Space dragons and carps have a temperature and heat resistance of INFINITY.

Pickle-Coding committed:
Whoops.

Fixes description of the plasma variable.
The plasma variable will now use MAXIMUM_STORED_PLASMA_MOLES when assigned.

JohnFulpWillard commented:
Plasmaflooding is usually round-ending and space dragons are currently very strong. If we do add this, I think we should make them cost more in dynamic at least (they're considered the same as abductors at the moment)

Pickle-Coding commented:
The plasma (at least in my testing) tend to burn up before they could be considered station wide. The heat might spread and stay longer though, but I haven't tested this on anything other than RuntimeStation.

IndieanaJones commented:
Could you post a video of this?

Pickle-Coding commented:
I'll post a video in a couple of hours when I have time.

newfren commented:
does the dragon still deal damage with the straight beam of fire or is it replaced by this?

Pickle-Coding commented:
It will still deal damage.

tralezab commented:
I'm unironically okay with this under the pretense that dragons get their dynamic adjusted accordingly, and the carp they summon don't just die to it right away. Your justification SHOULD be that it encourages space dragons to pick interesting, ATMOSPHERIC drop locations they can take advantage of pflooding

bluetshirtguy commented:
Atmos coder pr

IndieanaJones commented:
> If anything has to go tho, please remove it's armor penetration. Its stupid that half of our antagonists just ignore a good chunk of armor making an officer that spend time gearing up to counter a threat be the same as a random armorless assistant.

Bad idea, because guess who has the best fire resist and general armor on the station?

GuillaumePrata commented:
> Bad idea, because guess who has the best fire resist and general armor on the station? It's certainly not security.

Miners have stupid high values for armor while lavaland fauna have high values for armor penetration, making them the top dogs of the station while not helping them do their actual job. Literal noob trap for miners and power creep for antags.

We shouldn't let armor penetration exist unless it is needed as it nullifies the core of our sandbox, collecting the right items to use against threats.
Bypassing armor is disservice to Dragons as they have their own unique and more engaging ways to fight the crew but left clicking someone at melee range 3 times still is wins you the engagement.
Space Dragons shouldn't play like a Desword traitor.

JohnFulpWillard commented:
> Bad idea, because guess who has the best fire resist and general armor on the station? It's certainly not security.

It would require Miners to bother getting up and to the station to deal with it, which is at least a good thing becuase Miners are now interacting with the round in an actually meaningful way, while still doing their 'job' of hunting fauna.

Pickle-Coding committed:
Define and armour respect.

Adds PLASMA_TEMPERATURE define, local to space_dragon.dm, set to 2000. Replaces the values of "2000" in the TEMP parameter value in the space dragon's atmos spawn air proc, and replaces the 2000 value for the hotspot_expose exposed_temperature parameter value.

Space dragon dragon_fire_line will use apply_damages with 10 brute and 20 burn instead of applying fire loss directly.

Pickle-Coding commented:
Here is a shitty video I had to fit in 10MB. One thing I noticed is that it's a bit hard to see the dragon. I'm not sure how I would increase its visibility.
https://user-images.githubusercontent.com/58013024/194398188-b9fa4b44-574a-432f-a039-2a4bd7ce2663.mp4

Pickle-Coding commented:
I've set the damage to 10 brute and 20 burn, no longer ignoring armour. Complete fire resistance will block most of the damage.

Pickle-Coding committed:
Reduces antagonist event weight.

Reduces the weight of space dragon rolling to 5, from 7, by setting weight to 5 instead of 7 in the space_dragon round_event_control datum.

IndieanaJones commented:
> We shouldn't let armor penetration exist unless it is needed as it nullifies the core of our sandbox, collecting the right items to use against threats. Bypassing armor is disservice to Dragons as they have their own unique and more engaging ways to fight the crew but left clicking someone at melee range 3 times still is wins you the engagement. Space Dragons shouldn't play like a Desword traitor.

This PR also nerfs the fire line so it doesn't have a portion of damage that ignores armor. If this PR goes through, the only thing that can damage a drake armor miner *is* the dragon's bite. They'd be literally immune to both the direct flame attack and the plasma fire. You also have to realize that miners have the crusher, which does work on Space Dragon, along with self-heals to win a left-click engagement, the miner easily wins. And, if you remove the armor penetration, dragons can't harm miners anymore.

> It would require Miners to bother getting up and to the station to deal with it, which is at least a good thing becuase Miners are now interacting with the round in an actually meaningful way, while still doing their 'job' of hunting fauna.

There's a difference between helping take down a threat vs. being the only job that can take down a threat. Miners were already very good at hunting Space Dragons, but this PR pretty much bars almost anyone else from trying.

> If armor pen gets a more widespread nerf it might not be necessary for miners to have ridiculous armor. The flipside of this is that lavaland mobs would also need something to keep them from getting wrecked by security gear more than miners.

This is true, but that seems out of scope for this PR outside of removing the dragon's armor penetration, which again, leads to a scenario where the dragon literally cannot do anything to a miner except make him fall down for a bit.

> Here is a shitty video I had to fit in 10MB. One thing I noticed is that it's a bit hard to see the dragon. I'm not sure how I would increase its visibility. https://user-images.githubusercontent.com/58013024/194398188-b9fa4b44-574a-432f-a039-2a4bd7ce2663.mp4

That looks as nasty as I could've imagined it. I'm against this change, but it seems it might get merged for the funny haha moment so we can make SD barely ever spawn once people complain about this change in about a month from merging.

GuillaumePrata commented:
> You also have to realize that miners have the crusher, which does work on Space Dragon, along with self-heals to win a left-click engagement, the miner easily wins. And, if you remove the armor penetration, dragons can't harm miners anymore.

This PR also made the carps not be useless and killed by friendly fire.
If the dragon wants to go on a left click fiesta with a miner while they have AoE stun, an army of disposable carp minions + ignore the fact that Ash Drake armor is not space proof, they should lose as the miner is the one playing smart and using what they collected on the round for their advantage.

Pickle-Coding commented:
I don't think heat resistance prevents burn damage from other sources. I could disable friendly fire though. I assumed there was a reason for friendly fire, but I don't know.

Pickle-Coding commented:
> This is true, but that seems out of scope for this PR outside of removing the dragon's armor penetration, which again, leads to a scenario where the dragon literally cannot do anything to a miner except make him fall down for a bit.

The fire breath also does 10 brute damage. It's not a lot, but the dragon also has a 35 melee damage attack with 30 armour penetration, so it's not impossible. I think fire safety is a good thing for people to go for, as I can't see a lot of other antagonists have that as a counter, making a more unique engagement.

IndieanaJones commented:
> This PR also made the carps not be useless and killed by friendly fire. If the dragon wants to go on a left click fiesta with a miner while they have AoE stun, an army of disposable carp minions + ignore the fact that Ash Drake armor is not space proof, they should lose as the miner is the one playing smart and using what they collected on the round for their advantage.

Carps will still die when they get hit by the fire. What I was going to do was make them not take damage from it and speed up for a short duration when hit by the fire line to give synergy between the dragon and carps. Also, the space proof feature of drake armor doesn't matter, because it's a weakness which can be easily negated by pressure adaptation from genetics or the xenobio pressure resist potion. Also, a disposable army of carp minions doesn't matter when your armor nullifies the damage those carps can do.

> The fire breath also does 10 brute damage. It's not a lot, but the dragon also has a 35 melee damage attack with 30 armour penetration, so it's not impossible. I think fire safety is a good thing for people to go for, as I can't see a lot of other antagonists have that as a counter, making a more unique engagement.

The issue with this is that all of our fire safety gear has inherent slowdown (except drake armor) and the space dragon moves at regular carbon speed.

Pickle-Coding committed:
Fire breath no longer friendly fire.

Adds a "carp" in living.faction check that adds the living to the hit_list at the start of the for loop in dragon_fire_lane proc. This prevents things in the "carp" faction getting directly damaged by the fire breath.

Pickle-Coding commented:
> The issue with this is that all of our fire safety gear has inherent slowdown (except drake armor) and the space dragon moves at regular carbon speed.

As the space dragon is a heavy ruleset, it is expected to be a defining factor of the round and require lots of coordination to take down. Security officers could fire a stream of lasers to try and take the dragon down.

Alternatively security officers could not wear heavy fire resistant equipment, and instead have a team of atmospheric technicians, engineers, or anyone with a fire extinguisher to help prevent the fire from spreading and to extinguish the officers to slow down fire damage.

Jackal-boop commented:
i think this is a really cool change but even normal security has a hard time dealing with dragons, I think security and magnate modsuits need a buff to their protective properties in order to cope with this, when the HOS can be given "severe bruising" from a single bite while in a hos modsuit, it makes fighting them a gamble

Pickle-Coding committed:
Generalises get_stream_turfs()

Changes the /mob/living/simple_animal/hostile/space_dragon/proc/get_stream_turfs() proc to check for closed turfs, and whether the turf can share or not.

If it's a space turf, it will return if there are any atoms in it that fail the atom.can_atmos_pass() check, otherwise it will return if the turf has no adjacent atmos turfs.

Pickle-Coding committed:
Small comments cleanup.

Changes some comments.

Fikou commented:
@IndieanaJones miners only really have armor for fire stuff and melee stuff, what if the breath did energy damage?

IndieanaJones commented:
> @IndieanaJones miners only really have armor for fire stuff and melee stuff, what if the breath did energy damage?

I'm not sure if we have any heat-resistant armor with energy protection, but I do know that if you take the forced burn damage off the fire breath, SD will greatly struggle against engineering and mining cyborgs due to lacking any meaningful way to harm them. Energy damage will hurt them, but it does physical, which means that they can just weld as opposed to requiring wire repair from a roboticist.

Also, yeah, I get the point that a heavy midround should be scary, but this would make Space Dragon take wildly more effort to take down, much more than any of the midround antagonists forgoing xenomorphs with a good queen. I would actually compare this to blazing oil blob when they have a plasma fire on top of them, once that happens blob essentially gets a free win because there's not enough fire-retardant gear which can be used in combat to actually put up a decent fight. Again, from what I remember, all of the most common sources of heat protection come with a hefty slowdown since MODsuits got merged,  and the whole fire-fighting idea won't work because while you could put out part of the fire, the heat and pressure will still force players to be properly suited up to withstand the environment.

I'm just thinking if this gets merged, unless the crew contains a good miner or engineering/mining cyborgs, dragon will win because anyone who can withstand the inferno is going to be quite slow because of the gear they need to wear and a very easy target for a dragon and his minions who all move at base carbon speed. If the maintainers are dead-set on merging this PR, I'd at least ask to have it test-merged to see how it plays in practice. Things can sound very cool on paper, but end up absolutely lame in practice, and as of now I feel this is one of those things.

datorangebottle commented:
> I would actually compare this to blazing oil blob when they have a plasma fire on top of them, once that happens blob essentially gets a free win because there's not enough fire-retardant gear which can be used in combat to actually put up a decent fight.

blazing oil is perhaps one of the worst strains because it allows people to attack it with an extinguisher that hits huge amounts of blob at once. the second they bring in more water you're losing huge swathes of blob to people with an incredibly common tool that also fights the fire at the exact same time.

IndieanaJones commented:
> blazing oil is perhaps one of the worst strains because it allows people to attack it with an extinguisher that hits huge amounts of blob at once. the second they bring in more water you're losing huge swathes of blob to people with an incredibly common tool that also fights the fire at the exact same time.

Doesn't matter if the fire started off with enough fuel, the fire will spread so aggressively that fire extinguishers can't make a dent in it.

moocowswag commented:
> Plasmaflooding is usually round-ending and space dragons are currently very strong. If we do add this, I think we should make them cost more in dynamic at least (they're considered the same as abductors at the moment)

space dragons already technically are round ending because of the way carps are encouraged to avoid combat all together and sabotage the station and distract the crew as best as they can.

A carp will die in one hit from an energy gun and their ability to damage someone even if they manage to survive long enough to get into melee is extremely embarrassing to where they literally are better off pulling threats away from the portals by depressurizing and sabotaging most of the station by breaking airlocks, external windows, and firelocks in order to get people to try to deal with them instead.

bruhman8ghughug commented:
op as fuck, based as fuck, merge

san7890 commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -292,25 +320,31 @@
-		to_chat(L, span_userdanger("You're hit by [src]'s fire breath!"))
+		hit_list += living
+		living.apply_damages(brute = 10, burn = 20)
+		to_chat(living, span_userdanger("You're hit by [src]'s fire breath!"))
```
Could be nice to have this be a short bubble alert? Maybe something like "burned by dragon!" or something? Not sure.

LemonInTheDark commented on code/modules/events/ghost_role/space_dragon.dm:
```diff
@@ -1,7 +1,7 @@
 	name = "Spawn Space Dragon"
 	typepath = /datum/round_event/ghost_role/space_dragon
-	weight = 7
+	weight = 5
```
this is config isn't it?

LemonInTheDark replied:
so yaknow, talk to moth about this

Mothblocks replied:
this isnt dynamic either but these are basically completely unused outside of greenshifts

LemonInTheDark replied:
oh it... oh god you're right

LemonInTheDark commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -1,5 +1,15 @@
+/// The maximum amount of plasma that can get stored in moles.
+#define MAXIMUM_STORED_PLASMA_MOLES 800
+/// The maximum amount of plasma that can get released in a single fire breath in moles.
+#define MAXIMUM_PLASMA_MOLES_RELEASED 400
```
jesus

LemonInTheDark commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -259,31 +272,46 @@
+*/
+/mob/living/simple_animal/hostile/space_dragon/proc/get_stream_turfs(atom/target, range)
+	var/list/turfs = list()
+	for(var/turf/turf in line_target(0, range, target))
+		if(isclosedturf(turf))
+			return turfs
+		if(isspaceturf(turf))
+			for(var/atom/atom in turf)
+				if(!atom.can_atmos_pass(turf))
+					return turfs
+		else if(!TURF_SHARES(turf))
+			return turfs
+		turfs += turf
+	return turfs
```
this will fail if you breath into a 1x1 room

Pickle-Coding replied:
Is there a cheap way of handling that?

LemonInTheDark replied:
rather then checking TURF_SHARES
store the PREVIOUS turf we walked over, and use CAN_SHARE or whatever it is between those two
If there is no previous turf, ignore the check

san7890 commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -292,25 +320,31 @@
+ * * affected_volume - The number of turfs that got influenced by the fire stream.
  */
-/mob/living/simple_animal/hostile/space_dragon/proc/dragon_fire_line(turf/T)
+/mob/living/simple_animal/hostile/space_dragon/proc/dragon_fire_line(turf/turf, affected_volume, total_plasma_released)
```
you forgot to DMDoc what `total_plasma_released` is doing here

LemonInTheDark commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -259,31 +272,46 @@
-		for(var/obj/machinery/door/D in T.contents)
-			if(D.density)
-				return
+	var/list/stream_turfs = list()
+	stream_turfs = get_stream_turfs(target, range)
```
```suggestion
	var/list/stream_turfs = get_stream_turfs(target, range)
```

optimumtact commented:
great way to make an already shit mob shittier so I can remove it

LemonInTheDark commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -292,25 +320,31 @@
  * Arguments:
- * * turf/T - The turf to trigger the effects on.
+ * * turf/turf - The turf to trigger the effects on.
+ * * affected_volume - The number of turfs that got influenced by the fire stream.
```
volume is a bad name for this since it already has meaning, be more explicit

LemonInTheDark commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -292,25 +320,31 @@
-		if(L in hit_list)
+	if(isopenturf(turf))
+		var/turf/open/open_turf = turf
+		open_turf.atmos_spawn_air("plasma=[total_plasma_released / affected_volume];o2=[OXYGEN_RATIO * total_plasma_released / affected_volume];TEMP=[PLASMA_TEMPERATURE]")
```
why not just pass in the amount of plasma to spawn

LemonInTheDark commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -341,6 +375,10 @@
 
+/mob/living/simple_animal/hostile/space_dragon/get_status_tab_items()
+	. = ..()
+	. += "Plasma: [plasma]/[MAXIMUM_STORED_PLASMA_MOLES]"
```
it seems weird to store this in moles and not just shot counts

Pickle-Coding replied:
Most of the time the dragon would be firing without its full effectiveness. It would seem weird to say "0.2 shots" or something.

LemonInTheDark reviewed (changes requested):
This seems kinda too strong, specially if they get a new shot in at best 40 atmos ticks
Using gas like this can have unintended effects, because gas is meant to spread. Why is it suitable here?

san7890 commented on code/modules/mob/living/simple_animal/hostile/space_dragon.dm:
```diff
@@ -292,25 +320,31 @@
-		hit_list += M
-		M.take_damage(50, BRUTE, MELEE, 1)
+		hit_list += mecha
+		mecha.take_damage(50, BRUTE, MELEE, 1)
```
```suggestion
		mecha.take_damage(50, BRUTE, MELEE)
```

The fourth arg for `mecha.take_damage` is the sound effect, which defaults to TRUE (1) (at least for here). No need to write it again.

Pickle-Coding commented:
> > Here is a shitty video I had to fit in 10MB. One thing I noticed is that it's a bit hard to see the dragon. I'm not sure how I would increase its visibility. https://user-images.githubusercontent.com/58013024/194398188-b9fa4b44-574a-432f-a039-2a4bd7ce2663.mp4
> 
> 
> 
> is this clip accurate to the current state of the pr?
> 
> This looks like it'll make it miserable to fight dragons who opt to hold up in viro or the AI satellite 
> 
> that is a LOT of plasma, not sure how the crew / sec is meant to counter it in confined spaces

The dragon would be very vulnerable to lasers in a confined space. I could decrease the plasma, but that would reduce the everything is flaming aesthetic.

Pickle-Coding committed:
Update code/modules/mob/living/simple_animal/hostile/space_dragon.dm

Co-authored-by: LemonInTheDark <58055496+LemonInTheDark@users.noreply.github.com>

Pickle-Coding committed:
Update code/modules/mob/living/simple_animal/hostile/space_dragon.dm

Co-authored-by: san7890 <the@san7890.com>

Pickle-Coding commented:
> This seems kinda too strong, specially if they get a new shot in at best 40 atmos ticks
> 
> Using gas like this can have unintended effects, because gas is meant to spread. Why is it suitable here?

Dragons are a rather destructive and powerful creature. Letting the dragon change the atmosphere (both, the atmosphere and the atmosphere) seems like a good way to express its destructive power.

Pickle-Coding committed:
Reviews and stuff.

Adds turf_plasma_moles variable, which is just the plasma moles over the amount of turfs it iterates through. Passes turf_plasma_moles into the plasma_released parameter in the dragon_fire_line proc. Dragon_fire_line proc will just use plasma_released parameter for the amount of plasma and oxygen it spawns, instead of using total plasma parameter divided by the volume parameter every time.

Replaces to_chat with a balloon alert in the dragon_fire_line proc!

Pickle-Coding commented:
> > Here is a shitty video I had to fit in 10MB. One thing I noticed is that it's a bit hard to see the dragon. I'm not sure how I would increase its visibility. https://user-images.githubusercontent.com/58013024/194398188-b9fa4b44-574a-432f-a039-2a4bd7ce2663.mp4
> 
> 
> 
> is this clip accurate to the current state of the pr?
> 
> This looks like it'll make it miserable to fight dragons who opt to hold up in viro or the AI satellite 
> 
> that is a LOT of plasma, not sure how the crew / sec is meant to counter it in confined spaces

On second thought, it should actually be weaker than this. I tested again, and the plasma doesn't last very long as it's in a relatively oxygen rich environment.

I believe recording this caused lag, causing atmos subsystem to slow down, which doesn't account for delta time, but dragon fire breath does account for delta time, making it able to fire more plasma before the plasma gets burned. It's a similar problem to how atmos lag can cause the SM to reach a higher power level as emitters hit the SM more frequently before powerloss gets called.

I suppose I could make it not account for delta time, and make plasma generation more in sync, to make the sticky plasma problem less bad when atmos lags.

LemonInTheDark commented:
It's not a matter of delta time. The atmos subsystem is lagging, but not the dragon subsystem.
That's the point of the MC, but it's biting you here

Mothblocks commented:
@LemonInTheDark @san7890 If either of you are interested in this pull request please move forward on it or close

Pickle-Coding commented:
I'll open a new one if I can get atmos lag to not be such a big problem.

LemonInTheDark commented:
ftr I don't think this is a good idea, just cause atmos is a lot more of a longterm threat then a fire spurt. s a reason we don't do this with other stuff
